server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/model/Id.scala

```diff
@@ -19,15 +19,27 @@
 
 package org.apache.james.jmap.model
 
+import eu.timepit.refined.api.{Refined, Validate}
+import eu.timepit.refined.boolean.And
+import eu.timepit.refined.string.MatchesRegex
+import org.apache.james.jmap.model.Id.IdConstraints
+
+final case class NonNull()
+
+object NonNull {
+  implicit def matchesNonNull[Any]: Validate.Plain[Any, NonNull] =
+    Validate.fromPredicate(
+      value => value != null,
+      _ => "value cannot be null",
+      NonNull())
+}
```

Comment:
Is that necessary to test an ID is not null?

Reply:
No, it's for testing the String value inside `Id`

Reply:
Sounds complicated^^'

Reply:
I don't get why we need to specify it should be NotNull as MatchesRegex will fail anyway

Reply:
I had written a comment in my pull request that is passing null into the constructor causing NPE exception inside `MatchesRegex`. Technically, it fails to validate null value (at runtime)

Reply:
I tried to use it without it... Yes it doesn throw a NPE. Is it normal or not though? Don't we want it to throw? I'm still confused with the syntax (just tried and pushed something).

Bug tracker of Refined, regarding null: 
```
After giving this some more thoughts I propose to not handle this within refined.
Reasoning as follows:

    Automatic negation of predicates in the case of null breaks the usage of Not and would need more work in several places to check for the null case. This complicates the internal logic and also the code base.
    Checking for null should be the job of the developer. These errors usually appear when data from the "outside" is not properly checked at runtime. When using things like refined-circe this can be circumvented by using proper fields like Option[T] instead of T for possible null values.

It would be convenient if refined handled these cases but as mentioned it would complicate things.
```
So Refined doesn't care about null values it seems, and leaves the responsibility to the implementation to check it, right?

Reply:
I do understand your logic, with creating a NonNull predicate to treat that case not handled by Refined. But seems I feel it checks all the predicates, thus the NPE in MatchesRegex. Then I see two solutions:
* we use Refined, but still need to check that value is not null on our side (like you did with your require before I guess?)
* we define new predicate (like a new MatchesRegex) that will check null value too. But then is there really a point still using the lib if we redefine the predicates?

Reply:
I'll go for the second option: define a new type with null safe capability. So someone can provide me a really good reason to not care about null in Scala.

Reply:
Or as @chibenwa suggested, we don't check for null value

Reply:
> Or as @chibenwa suggested, we don't check for null value

Rethinking about this, our pojos are directly deserialized from user input. We need to make sure a json with a null field being deserialized will throw an IllegalArgumentException.  (rejecting invalid user input)

Maybe play-json allows doing it for us? Maybe we will need using `requires`? Only some tests will tell us...

Reply:
see https://github.com/fthomas/refined/issues/450#issuecomment-375655000

A nullable field should be expressed as an Option

Reply:
Play json will not validate a null value. It can only assign it to an Optional field otherwise it will not validate the json and return a JsError.
So you don't have to handle null

Reply:
Ok thanks.

Then... Let's not validate null fields, and keep null fiel validations in java facade if needed?

Reply:
then do you agree to merge #3202 @remk @mbaechler ? an I’ll take what you did with refined in a separed pr?